alphas/alpha_utils.py

```diff
@@ -758,3 +760,24 @@ def ite(x, y: pd.DataFrame, z: pd.DataFrame) -> pd.DataFrame:
         x.fillna(0).astype(int) * y + (~x.astype(bool)).fillna(0).astype(int) * z
     except:
         raise Exception("Data cannot be substituted, boolean may not be same dimensions as dataframes")
+
+def bscore(data: pd.Series, window: int) -> pd.Series:
+    """Bollinger score over the last 'window' days."""
+    rolling_mean = data.rolling(window=window).mean()
+    rolling_std = data.rolling(window=window).std()
+    return (data - rolling_mean) / rolling_std
+
+def returns(data: pd.Series, window: int) -> pd.Series:
+    """Computes returns over a given window."""
+    return data.pct_change(periods=window) + 1
+
+def csscale(data: pd.Series) -> pd.Series:
```

Comment:
could you point me to where you found the implementation for this fxn?

Reply:
csscale:csscale(x) = cross sectionally scale x over traded instruments such that the sum of abs(x) = 1

the rationale i have for this one is that ur dividing each element of the series by a sum of abs values of all the elements (bc u need the final series abs values to add to 1 per the requirements) 

eg [1,-2,3], u divide by 6, sum the abs values of [1/6,-1/3,1/2], gives u 1